recognition/include/pcl/recognition/impl/linemod/line_rgbd.hpp

```diff
@@ -43,176 +43,143 @@
-      stream.write (buffer, fsize);
-      SparseQuantizedMultiModTemplate sqmmt;
-      sqmmt.deserialize (stream);
+      if (temp_file.extension () == ".sqmmt")
+      {
+        std::ifstream stream;
+        int nr_templates;
 
-      linemod_.addTemplate (sqmmt);
-      object_ids_.push_back (object_id);
+        stream.open (temp_file.string (), std::ios::in | std::ios::binary);
+        pcl::read (stream, nr_templates);
+        PCL_DEBUG ("[pcl::LineRGBD::loadTemplates] Reading and parsing %d templates from SQMMT files.\n", nr_templates);
 
-      // Increment the offset for the next file
-      ltm_offset += (ltm_header.getFileSize ()) + (512 - ltm_header.getFileSize () % 512);
+        for (std::size_t i=0; i<nr_templates; ++i)
+        {
+          SparseQuantizedMultiModTemplate sqmmt;
+          sqmmt.deserialize (stream);
+          linemod_.addTemplate (sqmmt);
+        }
 
-      delete [] buffer;
+        stream.close ();
+      }
     }
-
-    if (io::raw_lseek(ltm_fd, ltm_offset, SEEK_SET) < 0)
-      break;
   }
 
-  // Close the file
-  io::raw_close(ltm_fd);
+  fs::remove_all (temp_dir);
 
   // Compute 3D bounding boxes from the template point clouds
-  bounding_boxes_.resize (template_point_clouds_.size ());
-  for (std::size_t i = 0; i < template_point_clouds_.size (); ++i)
+  bounding_boxes_.resize(template_point_clouds_.size());
+  for (std::size_t i = 0; i < template_point_clouds_.size(); ++i) 
   {
-    PointCloud<PointXYZRGBA> & template_point_cloud = template_point_clouds_[i];
-    BoundingBoxXYZ & bb = bounding_boxes_[i];
-    bb.x = bb.y = bb.z = std::numeric_limits<float>::max ();
-    bb.width = bb.height = bb.depth = 0.0f;
+    computeBoundingBox (i);
+  }
 
-    float center_x = 0.0f;
-    float center_y = 0.0f;
-    float center_z = 0.0f;
-    float min_x = std::numeric_limits<float>::max ();
-    float min_y = std::numeric_limits<float>::max ();
-    float min_z = std::numeric_limits<float>::max ();
-    float max_x = -std::numeric_limits<float>::max ();
-    float max_y = -std::numeric_limits<float>::max ();
-    float max_z = -std::numeric_limits<float>::max ();
-    std::size_t counter = 0;
-    for (std::size_t j = 0; j < template_point_cloud.size (); ++j)
-    {
-      const PointXYZRGBA & p = template_point_cloud.points[j];
+  return (true);
+}
 
-      if (!std::isfinite (p.x) || !std::isfinite (p.y) || !std::isfinite (p.z))
-        continue;
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointXYZT, typename PointRGBT> void
+pcl::LineRGBD<PointXYZT, PointRGBT>::computeBoundingBox (int i)
+{
+  PointCloud<PointXYZRGBA>& template_point_cloud = template_point_clouds_[i];
+  BoundingBoxXYZ& bb = bounding_boxes_[i];
+  bb.x = bb.y = bb.z = std::numeric_limits<float>::max();
+  bb.width = bb.height = bb.depth = 0.0f;
+
+  float center_x = 0.0f;
```

Comment:
Can you split this into another PR? This is a chance to refactor this code to use Eigen instead of independent variables. We can focus on just the loading improvements in this PR